src/status_im/ui/screens/wallet_connect/session_proposal/views.cljs

```diff
@@ -0,0 +1,205 @@
+  [react/touchable-opacity {:on-press on-press}
+   [react/view (styles/toolbar-container background-color)
+    [quo/text {:color :inverse
+               :weight :medium
+               :style styles/toolbar-text}
+     text]
+    [icons/icon
+     :main-icons/chevron-down
+     {:color  (:text-05 @colors/theme)
+      :width  chevron-icon-container-width
+      :height chevron-icon-container-height}]]])
+
+(def show-account-selector? (reagent/atom false))
+
+(defn render-account [{:keys [address name color] :as account} _ _ {:keys [selected-account on-select]}]
+  (let [account-selected? (= (:address @selected-account) address)]
+    [react/touchable-without-feedback {:on-press #(do
+                                                    (reset! selected-account (merge {} account))
+                                                    (when on-select (on-select)))}
+     [react/view (styles/account-container color account-selected?)
+      [quo/text {:color :inverse
+                 :weight (if account-selected? :medium :regular)}
+       name]]]))
+
+(defn account-selector [accounts selected-account on-select]
+  [react/view styles/account-selector-container
+   [quo/text {:size :small} (i18n/label :t/select-account)]
+   [list/flat-list {:data        accounts
+                    :key-fn      :address
+                    :render-fn   render-account
+                    :render-data {:selected-account selected-account
+                                  :on-select on-select}
+                    :horizontal  true
+                    :shows-horizontal-scroll-indicator false
+                    :extra-data @selected-account
+                    :style styles/account-selector-list}]])
+
+(defn account-picker [accounts selected-account {:keys [on-press on-select]}]
+  (if (> (count accounts) 1)
+    [react/view {:style styles/account-selector-wrapper}
+     [account-selector accounts selected-account on-select]]
+    [react/touchable-opacity {:style styles/single-account-container}
+     [toolbar-selection {:text (:name @selected-account)
+                         :background-color (:color @selected-account)
+                         :on-press on-press}]]))
+
+(defview success-sheet-view [{:keys [topic]}]
+  (letsubs [visible-accounts [:visible-accounts-without-watch-only]
+            dapps-account [:dapps-account]
+            sessions [:wallet-connect/sessions]
+            managed-session [:wallet-connect/session-managed]]
+    (let [{:keys [peer state] :as session} (first (filter #(= (:topic %) topic) sessions))
+          {:keys [accounts]} state
+          {:keys [metadata]} peer
+          {:keys [name icons]} metadata
+          icon-uri (when (and icons (> (count icons) 0)) (first icons))
+          address (last (string/split (first accounts) #":"))
+          account (first (filter #(= (:address %) address) visible-accounts))
+          selected-account (reagent/atom account)]
```

Comment:
i think we still don't have any agreement on how to name atoms, but it's better to somehow show that its an atom, `selected-account-atom` would work for me

Reply:
why do you need to wrap it with the atom btw ?

Reply:
We need to get the session account and wrap it with the atom so user can select another one and pass it to extraData props of flat list so Flat List re-renders when value of the atom changes.